Dockerfile

```diff
@@ -53,12 +53,20 @@ RUN make rclone
 # Build restic
 FROM golang-builder as restic-builder
 
-COPY /mover-restic/restic ./restic
-COPY /mover-restic/minio-go ./minio-go
+ARG RESTIC_VERSION="v0.15.1"
+ARG RESTIC_GIT_HASH="7d4b7ad9cb45d6801df96fcf61f876d580416fa5"
 
+RUN git clone --depth 1 -b ${RESTIC_VERSION} https://github.com/restic/restic.git
 WORKDIR /workspace/restic
 
-RUN go run build.go
+# Make sure the restic version tag matches the git hash we're expecting
+RUN /bin/bash -c "[[ $(git rev-list -n 1 HEAD) == ${RESTIC_GIT_HASH} ]]"
+
+# HACK: Apply our patch to restic to enable FIPS mode
+RUN find . -name '*.go' -exec sed -ri 's|github.com/minio/sha256-simd|crypto/sha256|' {} \; && go mod tidy
```

Comment:
Is this step still necessary?  Unfortunately downstream we are not able to edit source before building - so we'd need to keep vendoring if we want to do this.  I'm hoping the fips build tag will solve the minio-go issues for fips, but now I'm uncertain (I'm struggling to remember the details of all this) if sha256-simd is also used directly by restic for some operations.

Reply:
Yes, restic still uses `sha256-simd` to compute hashes internally. 
The clean solution would be to open a pull request for fips compatibility in `restic` if a tag is given, similar to https://github.com/minio/minio-go/pull/1749. Then, no patching would be necessary.
I ended up using another backup solution for my project, so I won't continue working on this.

Reply:
/close

Based on all the discussion we're going to close this for the moment.